File: Backend/KebunJio/src/main/java/iss/nus/edu/sg/sa4106/KebunJio/Models/ActivityLog.java
package iss.nus.edu.sg.sa4106.KebunJio.Models;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonFormat;

@Document(collection = "ActivityLogs")
public class ActivityLog {
	
	@Id
    private String id;
    private String userId;
    private String plantId;
    private String activityType;
    private String activityDescription;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;
    
	public ActivityLog() {}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getUserId() {
		return userId;
	}

	public void setUserId(String userId) {
		this.userId = userId;
	}

	public String getPlantId() {
		return plantId;
	}

	public void setPlantId(String plantId) {
		this.plantId = plantId;
	}

	public String getActivityType() {
		return activityType;
	}

	public void setActivityType(String activityType) {
		this.activityType = activityType;
	}

	public String getActivityDescription() {
		return activityDescription;
	}

	public void setActivityDescription(String activityDescription) {
		this.activityDescription = activityDescription;
	}

	public LocalDateTime getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(LocalDateTime timestamp) {
		this.timestamp = timestamp;
	}
	
	
    
    

}

File: Mobile/app/src/main/java/iss/nus/edu/sg/sa4106/kebunjio/data/ActivityLog.kt
package iss.nus.edu.sg.sa4106.kebunjio.data

import iss.nus.edu.sg.sa4106.kebunjio.HandleNulls.Companion.ifNullString
import org.json.JSONObject
import java.io.Serializable

data class ActivityLog(
    val id: String,
    val userId: String,
    val plantId: String?,
    val activityType: String,
    val activityDescription: String, // can be blank
    val timestamp: String


) : Serializable {
    public fun dataIsGood(): Boolean {
        return (!plantId.equals("") &&
                !activityType.equals("") &&
                !timestamp.equals(""))
    }

    companion object {
        fun getFromResponseObject(responseObject: JSONObject): ActivityLog {
            val id = ifNullString(responseObject.getString("id"))
            val userId = ifNullString(responseObject.getString("userId"))
            val plantId = responseObject.getString("plantId")
            val activityType = ifNullString(responseObject.getString("activityType"))
            val activityDescription = ifNullString(responseObject.getString("activityDescription"))
            val timestamp = ifNullString(responseObject.getString("timestamp"))
            return ActivityLog(id, userId, plantId, activityType, activityDescription, timestamp)
        }
    }
}
